test/components/submission/filters.spec.js

```diff
@@ -462,4 +462,23 @@ describe('SubmissionFilters', () => {
       });
     });
   });
+
+  it('should not send Submission OData request on navigating to another route after filter', () => {
+    testData.extendedForms.createPast(1);
+
+    return loadComponent('reviewState=%27approved%27')
+      .beforeEachResponse((_, { url }) => {
+        if (url.includes('.svc')) {
+          const filter = relativeUrl(url).searchParams.get('$filter');
+          filter.should.equal("(__system/reviewState eq 'approved')");
+        }
+      })
+      .afterResponses(component => {
+        const filters = component.getComponent(SubmissionFilters).props();
+        filters.reviewState.should.eql(["'approved'"]);
+      })
+      .complete()
+      .route('/projects/1/forms/f/draft/testing')
```

Comment:
This navigation would change the page and render a new route component. Given that, I think we should use `load()` instead of `loadComponent()`. That way, the route component can change.